improve-service-processing/README.md

```diff
@@ -0,0 +1,73 @@
+  [2022-04-12 21:44:07,538: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:fedora-rawhide-x86_64': SRPM build is in progress...
+  [2022-04-12 21:44:07,870: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:fedora-35-aarch64': SRPM build is in progress...
+  [2022-04-12 21:44:08,215: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:fedora-36-x86_64': SRPM build is in progress...
+  [2022-04-12 21:44:08,541: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:centos-stream-9-aarch64': SRPM build is in progress...
+  [2022-04-12 21:44:08,893: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:fedora-34-s390x': SRPM build is in progress...
+  [2022-04-12 21:44:09,227: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:epel-8-x86_64': SRPM build is in progress...
+  [2022-04-12 21:44:09,576: DEBUG/ForkPoolWorker-1] Setting Github status check 'in_progress' for check 'rpm-build:fedora-rawhide-aarch64': SRPM build is in progress...
+  ```
+
+  - we probably can't do much about it
+
+- processing event when there is no config file
+  - finding out there is no config file takes few seconds (on average around 2, but this depends probably on the repo size,
+    I have seen also a log with 4 seconds duration)
+- the target expansion is also done more times during one processing, we could improve it (using properties and setting build/test targets only once),
+  but the process takes milliseconds, so this is not problematic from a time perspective
+
+## Suggestions for improvements
+
+#### Can we uninstall a repo? - Not receive events, after some period of time when we didn’t produce any meaningful work.
+
+- the documentation provides info on how to list installations, get a specific installation, delete, suspend it
+  (when a GitHub App is suspended, the app's access to the GitHub API or webhook events is blocked for that account)
+  https://docs.github.com/en/rest/reference/apps
+- we could have a periodic job which would check the installations and if an installed repo does not contain a config file
+  after some time limit (e.g. a month after installation), uninstall it
+  - if the app is installed on a namespace - at least one repo with a config file?
+
+#### Add higher prio to tasks from repos who are active users
+
+- in service we could get the repo name and check it in DB and accordingly route the tasks to different queues
+  (as we do currently for long and short running tasks)
+- the info about "active" users would be updated in the worker tasks
+  - do we want to have it only binary?
+    1. run at least one job for the repo
+    2. no jobs run
+  - or something more granular?
```

Comment:
Maybe. Is there any built-in support for priorities in celery?

It would be nice to be able to set it runtime. I can imagine various rules (and some a bit questionable) that can potentially be combined:
* Avoid overloading service by one project. (Decrease priority if we have at least X jobs already running for this namespace/project/PR.)
* Historically active users. (Increase priority if a project has at least X jobs during the last week/month/...)
* Decrease priority of notification events (we already have two queues for this).
* Decrease priority for some triggers -- e.g. Decrease priority if the trigger is push/release.
* Increase/Decrease priority for some projects -- e.g. set it in the service-config. (Test projects can have lower priority, some privileged projects higher.)

But having too many rules can lead to issues with starvation so we don't need to overthink this. At least from the start.

Reply:
great suggestions, I added some into the document and also added info about Celery priorities